carbon-design-system/carbon
# feat : Implement codemod to rename slug prop to aiLabel for IBM Products components

Closes [6307](https://github.com/carbon-design-system/ibm-products/issues/6307)

This PR updates the @carbon/upgrade package with a new transform `ibm-products-slug-to-ailabel-prop`, to update the `slug` prop in `ibm products components` `Tearsheet`, `SidePanel`, `ExpressiveCard` and `ProductiveCard` to `AILabel `.

#### Changelog

**New**

- New transform file `packages/upgrade/transforms/ibm-products-slug-to-ailabel-prop.js`
- Include fixtures in the `packages/upgrade/transforms/__testfixtures__/ibm-products-slug-to-ailabel-prop.input.js` and `packages/upgrade/transforms/__testfixtures__/ibm-products-slug-to-ailabel-prop.output.js` files
- Include tests in `packages/upgrade/transforms/__tests__/ibm-products-slug-to-ailabel-prop-test.js`


**Changed**

- Update `packages/upgrade/src/upgrades.js` with a new migration entry `ibm-products-slug-to-ailabel-prop`

#### Testing / Reviewing

- Pulling down this PR
- Run cd packages/upgrade
- Run yarn build to build the CLI
- Run `./bin/carbon-upgrade.js` migrate list and verify a new entry `ibm-products-slug-to-ailabel-prop` present
- Run `npx jest transforms/__tests__/ibm-products-slug-to-ailabel-prop-test.js` and verify test result
